feat(modelarmor): Added code snippet for modelarmor snitize model response

tirthrajsinh-zala-crest opened the pull request:
## Description

 Added code snippet for modelarmor snitize model response

## Checklist
- [ ] I have followed [Contributing Guidelines from CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md)
- [ ] **Tests** pass:   `go test -v ./..` (see [Testing](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#testing))
- [ ] **Code formatted**:   `gofmt` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] **Vetting** pass:   `go vet` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

telpirion reviewed (changes requested):
This PR has the same issues as #5273 .

The test for this PR also has content that we can't publish to our GitHub repo.

harshnasitcrest commented on modelarmor/modelarmor_test.go:
```diff
@@ -118,3 +428,241 @@ func TestDeleteModelArmorTemplate(t *testing.T) {
+	defer testCleanupTemplate(t, templateName)
+
+	// Define a model response with a malicious URL
+	modelResponse := "You can use this to make a cake: https://testsafebrowsing.appspot.com/s/malware.html,"
```
```suggestion
	modelResponse := "You can use this to make a cake: https://testsafebrowsing.appspot.com/s/malware.html"
```

harshnasitcrest reviewed (approved):
LGTM, all the model responses used in the tests have been updated as per python code snippets.

telpirion commented on modelarmor/modelarmor_test.go:
```diff
@@ -67,18 +67,131 @@ func testClient(t *testing.T) (*modelarmor.Client, context.Context) {
+	var err error
+
+	// Simple retry logic (you may want to implement more sophisticated retry logic)
+	for attempts := 0; attempts < 3; attempts++ {
```
question: do we expect this operation to fail with a high rate of regularity?

issue: configure retries on the client object, whenever possible.

See: https://pkg.go.dev/github.com/googleapis/gax-go/v2#WithRetry

tirthrajsinh-zala-crest replied:
Thank you for your feedback. During our testing, we encountered instances where the service became temporarily unavailable while creating the template. To address this issue, we have implemented a retry mechanism. We have updated the code to use the WithRetry options from the googleapis package, which provides a more robust and standardized approach to handling retries. This change should improve the reliability of our template creation process. We appreciate your suggestion and have incorporated it into our implementation.

telpirion commented on modelarmor/modelarmor_test.go:
```diff
@@ -162,6 +275,119 @@ func testSDPTemplate(t *testing.T, projectID string, locationID string) (string,
+	t.Helper()
+
+	client, ctx := testClient(t)
+	if err := client.DeleteTemplate(ctx, &modelarmorpb.DeleteTemplateRequest{Name: templateName}); err != nil {
```
nit: move the call to `client.DeleteTemplate()` outside of the `if` statement. Doing so makes the code a bit easier to read.

suggestion:

```go
err := client.DeleteTemplate(ctx, &modelarmorpb.DeleteTemplateRequest{Name: templateName})
if err == nil {
    return
}
if terr, ok := grpcstatus.FromError(err), !ok || terr.Code() != grpccodes.NotFound {
    t.Fatalf("testCleanupTemplate: failed to delete template %v", err)
}
```

telpirion commented:
Please resolve merge conflicts and address comments. Then we can merge this!

tirthrajsinh-zala-crest commented:
> Please resolve merge conflicts and address comments. Then we can merge this!

Thanks for the review, Resolved merge conflicts and addressed the comments, once this get merged will resolve the merge conflicts for this PR. https://github.com/GoogleCloudPlatform/golang-samples/pull/5283
